Lib/subprocess.py

```diff
@@ -2031,7 +2031,7 @@ def _wait(self, timeout):
             if self.returncode is not None:
                 return self.returncode
 
-            if timeout is not None:
+            if timeout is not None and timeout > 0:
```

Comment:
_wait() with timeout <= 0 must **not** disable timeouts. That is a semantic behavior change and would cause issues in fuzzy elapsed time for timeout calculations to wind up not having a timeout in wait at all.

The goal of this ._wait implementation is to **always** call waitpid at least once before deciding if TimeoutExpired should be raised.